Feature/122 enhance raw transaction view

ahoseinian opened the pull request:
### Description
- Refactor Transaction context to use better state management
- Add Preview page to transactions to see details and fees applied

part of #122 


https://user-images.githubusercontent.com/1168983/121893458-4416f400-cd1e-11eb-886e-9b4dcc23910b.MP4

ahoseinian committed:
Add HashBlock component

fixes #122

ahoseinian committed:
Update HashBlock.tsx

fixes #122

ahoseinian committed:
UPDATE TxContext to use better state management

fixes #122

ahoseinian committed:
Update Podfile.lock

fixes #122

ahoseinian committed:
Add preview screen

fixes #122

ahoseinian committed:
Update preview screen

fixes #122

ahoseinian committed:
Show fees

fixes #122

ahoseinian commented:
@franjs @geastwood can I get a review here?

silva-fj commented on src/context/TxContext/PreviewStep.tsx:
```diff
@@ -0,0 +1,89 @@
+  transactionTitle: string;
+  transactionInfo: string;
+  payload: SignerPayloadJSON;
+  params: unknown;
```
should `params` be an array?

ahoseinian replied:
yes, I have changed it 👍

silva-fj commented on src/context/TxContext/index.tsx:
```diff
@@ -86,8 +62,12 @@ function TxContextProvider({children}: PropTypes) {
+          transaction.paymentInfo(address).then((info) => {
+            dispatch({
+              type: 'PREVIEW',
+              payload: {payload, params, title, description, partialFee: info.partialFee.toNumber()},
```
how about `payload: {txPayload, ....` ?

silva-fj commented on src/context/TxContext/index.tsx:
```diff
@@ -102,63 +82,76 @@ function TxContextProvider({children}: PropTypes) {
-            );
+              } = eventRecord;
+
+              if ((error as any).isModule) {
```
why `any`? There should be a type for this error

ahoseinian replied:
yes you are right, I found an example and did the same 
https://github.com/polkadot-js/apps/blob/d1389052c43c24585ce87b17da8a82dc6e8294e9/packages/react-components/src/Status/Queue.tsx#L88

silva-fj commented on src/context/TxContext/index.tsx:
```diff
@@ -102,63 +82,76 @@ function TxContextProvider({children}: PropTypes) {
+
+              if ((error as any).isModule) {
+                // for module errors, we have the section indexed, lookup
+                const decoded = api.registry.findMetaError((error as any).asModule);
```
same

silva-fj commented on src/presentational/IdentityInfoForm.tsx:
```diff
@@ -3,32 +3,26 @@ import {View, StyleSheet, ScrollView} from 'react-native';
 
 type IdentityInfoFormProps = {
-  onSubmit: (identityInfo: IdentityInfo) => void;
+  onSubmit: (identityInfo: SubmitIdentityPayload) => void;
```
NIT: `IdentityInfo` or `IdentityPayload` would cleaner IMO

silva-fj commented on src/presentational/TxPayloadQr.tsx:
```diff
@@ -1,39 +1,40 @@
+      return;
+    }
+    // limit size of the transaction
+    const isQrHashed = payload.method.length > 5000;
```
Can you move the number to a constant and remove the comment?

Like:
`const TX_SIZE_LIMIT = 5000;`

silva-fj commented on src/screen/MyIdentityScreen/RequestJudgement.tsx:
```diff
@@ -40,7 +40,14 @@ function RequestJudgement({display, address, detail}: PropTypes) {
+          txMethod: 'identity.requestJudgement',
+          params: [index, fee],
+          title: `Sending transaction identity.requestJudgement`,
+          description: 'TODO',
```
what's TODO?

ahoseinian replied:
forgot to fix this, it's now `Request a judgement from a registrar.`

ahoseinian committed:
Fix PR comments

- https://github.com/litentry/litentry-app/pull/123#discussion_r651789103
- https://github.com/litentry/litentry-app/pull/123#discussion_r651792278
- https://github.com/litentry/litentry-app/pull/123#discussion_r651794901
- https://github.com/litentry/litentry-app/pull/123#discussion_r651795145
- https://github.com/litentry/litentry-app/pull/123#discussion_r651802321
- https://github.com/litentry/litentry-app/pull/123#discussion_r651805260
- https://github.com/litentry/litentry-app/pull/123#discussion_r651808683

fixes #122

silva-fj reviewed (approved):
Good improvements 👍🏼
